lib/remove-empty.js

```diff
@@ -0,0 +1,27 @@
+var Plugin = require('broccoli-plugin');
+var walkSync = require('walk-sync');
+var fs = require('fs');
+var path = require('path');
+
+module.exports = RemoveFiles;
+
+RemoveFiles.prototype = Object.create(Plugin.prototype);
+RemoveFiles.prototype.constructor = RemoveFiles;
+function RemoveFiles(inputNode, options) {
+  options = options || {};
+  Plugin.call(this, [inputNode], {
+    annotation: options.annotation
+  });
+}
+
+RemoveFiles.prototype.build = function() {
+  for (var i = 0; i < this.inputPaths.length; i++) {
```

Comment:
Again sorry there is not yet a functioning nicer high level API for this, hopefully after my next few speaking engagements I will again have free time to finish up that effort. I feel the current state is extremely crappy and largely because I haven't had a chance to finish my effort yet...